torch/csrc/jit/codegen/cuda/ir_container.h

```diff
@@ -128,11 +140,32 @@ class TORCH_CUDA_CU_API IrContainer : public PolymorphicBase {
   // something like check if an Expr is in this container
   std::unordered_set<Expr*> exprs_;
 
+  // Used to implement a generic "inContainer" that can be passed an invalid
+  // pointer. Specifically a pointer to a Statement owned by another container
+  // that has been freed. We can't check normally with the unordered_sets we
+  // already have because it would require a const_cast from a constant
+  // expr/val, or a dynamic cast from a Statement.
+  std::unordered_set<void*> raw_ptrs_;
```

Comment:
I think this is clear and simple to understand, so I'm not proposing a change, but just for reference, I believe it would have worked if C-style cast was used.

```
 bool IrContainer::inContainer(const Statement* stmt) const {
+#if 0
   const void* const_void = (const void*)(stmt);
   void* nonconst_void = const_cast<void*>(const_void); // NOLINT
   if (raw_ptrs_.find(nonconst_void) == raw_ptrs_.end()) {
     return false;
   }
+#else
+  auto expr = (Expr*)(stmt);
+  auto val = (Val*)(stmt);
+  TORCH_INTERNAL_ASSERT(val == stmt);
+  TORCH_INTERNAL_ASSERT(expr == stmt);
+  if (exprs_.find(expr) == exprs_.end() &&
+      vals_.find(val) == vals_.end()) {
+    return false;
+  }
+#endif
```